packages/client-core/src/util/upload.tsx

```diff
@@ -37,6 +37,71 @@ import { RethrownError } from './errors'
 export type CancelableUploadPromiseReturnType<T = any> = { cancel: () => void; promise: Promise<T | T[]> }
 export type CancelableUploadPromiseArrayReturnType<T = any> = { cancel: () => void; promises: Array<Promise<T | T[]>> }
 
+const getFileKeys = (files: Array<File> | File) => {
+  const keys = [] as string[]
+  if (Array.isArray(files)) {
+    files.forEach((file) => {
+      keys.push(file.name)
+    })
+  } else {
+    keys.push(files.name)
+  }
+
+  return keys
+}
+
+export const useUploadingFiles = (): [number, number, number] => {
+  const fileUploadState = useMutableState(FileUploadState).value
+  const values = Object.values(fileUploadState)
+  const total = values.length
+  const completed = values.reduce((prev, curr) => (curr === 1 ? prev + 1 : prev), 0)
+  const sum = values.reduce((prev, curr) => prev + curr, 0)
+  const progress = sum ? (sum / total) * 100 : 0
+  return [completed, total, progress]
```

Comment:
better to return an object for easier usage of this
```suggestion
  return {completed, total, progress}
```